### PR TITLE
test(go): expand backend unit test coverage

### LINKED ISSUE #507: test(go): expand unit test coverage from 3 to 15+ test files
**Parent:** #503
**Priority:** 🟠 Strategic

**Affected:** `backend-go/internal/` (113 Go source files); currently only 3 test files exist: `auth/`, `password/`, `server/`

**Why:** Only ~2.6% of backend Go modules have unit tests. Most domain packages (debts, equity_grants, accounts, transactions, retirement, salaries, snapshots, goals) lack test coverage. This increases risk of regressions when refactoring handler/store logic.

**Impact:** High - bugs in untested domain logic can silently break endpoints.
**Contagion:** High - every new feature or refactor touches multiple domain packages.
**Effort:** Days - 10+ new test files (e.g., `debts_store_test.go`, `accounts_handler_test.go`).
**Business Alignment:** High - domain logic (net worth, tax calculations, asset allocation) must be reliable.

**Fix:**
1. Identify top 3 critical domain packages by LOC + usage: e.g., `snapshots`, `accounts`, `debts` (first targets for unit tests).
2. Create a test file per package-level concern: `internal/debts/store_test.go`, `internal/debts/validation_test.go`.
3. Write fixtures and subtests for CRUD, edge cases, and error handling.
4. Run `go test ./...` and aim for >80% statement coverage on critical paths.
5. Add coverage thresholds to CI (optional but recommended).

**Test approach:** Run `cd backend-go && go test ./... -v` and check coverage: `go test ./... -cover | grep -E "total|debts|accounts"`.

**Estimated:** 3-5 days

### PR DESCRIPTION
**Pull request by Automaat**
## Motivation

Pre-this-PR, only 3 of the ~25 backend-go packages had test files (`auth/token`, `auth/password`, `server`). Every domain endpoint group depended entirely on the bb-tests integration suite for regressions, leaving validation/parsing logic — small, fast, deterministic — exercised only end-to-end against a live Postgres.

## Implementation information

Added 13 new `*_test.go` files covering pure logic that doesn't need a database. Each file targets one obvious slice of its package:

- `accounts/validation_test.go` + `handler_test.go` (request/patch parsing, `isoNaive`/`pyFloat`, `parseIDParam`)
- `snapshots/validation_test.go` (request/patch parsing, value entries)
- `goals/validation_test.go` + `projection_test.go` (validate, patch, `projectHitDate`, `addMonths`, `daysIn`)
- `assets/validation_test.go` (name helpers, isNull)
- `config/validation_test.go` (every `validate()` branch incl. allocation sum, age relationship)
- `salaries/validation_test.go` (create/patch, future-date guard)
- `bonus_events/validation_test.go` (create/patch, notes / owner-clearing semantics)
- `auth/validation_test.go` (username, password, optional name, PPK bounds)
- `cpi/inflation_test.go` (`CumulativeIndex`, `IndexAtDate` clamp + interpolate + missing-year, `AdjustWithIndex`)
- `fx/fx_test.go` (`ToPLN` currency/rate handling, `truncateDay`)
- `zus/validation_test.go` (`buildInputs` happy + each guard, salary history, range helpers)

Result: 16 test files. `go test ./...` and `golangci-lint run ./...` both clean.

Closes #507

> Changelog: test(go): expand backend unit test coverage

### CODE REVIEW

File: backend-go/internal/fx/fx_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	in := time.Date(2026, 5, 23, 13, 45, 30, 999, time.UTC)
+	got := truncateDay(in)
+	want := time.Date(2026, 5, 23, 0, 0, 0, 0, time.UTC)
+	if !got.Equal(want) {
+		t.Fatalf("got %s want %s", got, want)
+	}
+}
+
+func TestTruncateDayForcesUTC(t *testing.T) {
+	loc, _ := time.LoadLocation("Europe/Warsaw")
```
[the remark this answers is not in the document]
Comment (Automaat) [reply backend-go/internal/fx/fx_test.go]: Switched to `time.FixedZone` so the test no longer relies on tzdata being present on the runner.

File: backend-go/internal/accounts/validation_test.go [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+	}
+}
+
+func TestOptionalDecimalParsesAndFails(t *testing.T) {
+	good := rawJSON(t, map[string]any{"x": 12.345})
+	d, vErr := optionalDecimal(good, "x")
+	if vErr != nil || d == nil || !d.Equal(decimal.RequireFromString("12.345")) {
+		t.Fatalf("good parse failed: %+v err=%+v", d, vErr)
+	}
+	bad := map[string]json.RawMessage{"x": json.RawMessage("not-a-number")}
```
[the remark this answers is not in the document]
Comment (Automaat) [reply backend-go/internal/accounts/validation_test.go]: Good point — switched to constructing the raw map via the rawJSON helper so the invalid case is a quoted string (valid JSON token, just not a parseable decimal).
